### PR TITLE
Adapt code to orchestrator 2.0

### PR DESCRIPTION
**Pull request by hanstrompert**

### CODE REVIEW

File: workflows/core_link/create_core_link.py [hunk 1]
```diff
@@ -35,17 +37,17 @@
...
 
         node_subscription_id_a: node_selector("NodesEnumA")  # type:ignore # noqa: F821
         node_subscription_id_b: node_selector("NodesEnumB")  # type:ignore # noqa: F821
 
-        @validator("node_subscription_id_b", allow_reuse=True)
-        def separate_nodes(cls, v: str, values: dict, **kwargs):
-            if v == values["node_subscription_id_a"]:
-                raise AssertionError("node B cannot be the same as node A")
-            return v
+        @field_validator("node_subscription_id_b")
```
Comment (mrijk) [inline workflows/core_link/create_core_link.py:45]: I think for now this is ok. An improvement might be to have an annotated type NodeSubscriptionA and NodeSubscriptionB, where you could add this validator to NodeSubscriptionB. Alternatively If you want to decouple the two types, an model_validator might make sense.

File: workflows/port/create_port.py [hunk 2]
```diff
@@ -51,15 +52,14 @@ class Config:
...
     class CreatePortForm(FormPage):
-        class Config:
-            title = product_name
+        model_config = ConfigDict(title=product_name)
 
         # organisation: OrganisationId
 
         port_settings: Label
 
         port_ims_id: free_port_selector(node_subscription_id, speed)  # type:ignore
```
Comment (mrijk) [inline workflows/port/create_port.py:61]: Eventually we don't want to have function calls in Form's, just types. 

So you would have something like 'PortImsId = free_port_selector(node_subscription_id, speed)'. There is some trickery involved here to make mypy happy, unless you can live with loads of type ignores.

In some places in surf orchestrator we are already introducing this new approach, but maybe keep it here for now untill that has fully crystalized.

File: workflows/shared.py [hunk 3]
```diff
@@ -107,34 +113,26 @@ def free_port_selector(node_subscription_id: UUIDstr, speed: int, enum: str = "P
...
-        data = SummaryData(**summary_data)
-
+def summary_form(product_name: str, summary_data: SummaryData) -> Generator:
     class SummaryForm(FormPage):
-        class Config:
-            title = f"{product_name} summary"
+        model_config = ConfigDict(title=f"{product_name} summary")
 
-        product_summary: ProductSummary
+        product_summary: migration_summary(summary_data)  # type: ignore
```
Comment (mrijk) [inline workflows/shared.py:120]: Most places in surf orchestrator we introduced a type 'MigrationSummary', something along the lines of

`ProductSummary = migration_summary(sumary_data)`

To make mypy happy this involves some casting

**Review by mrijk — APPROVED** [review]

File: workflows/core_link/create_core_link.py [hunk 4]
```diff
@@ -34,35 +38,41 @@
...
+
     class SelectNodes(FormPage):
-        class Config:
-            title = f"{product_name} - node A and B"
+        model_config = ConfigDict(title=f"{product_name} - node A and B")
 
-        node_subscription_id_a: node_selector("NodesEnumA")  # type:ignore # noqa: F821
-        node_subscription_id_b: node_selector("NodesEnumB")  # type:ignore # noqa: F821
+        node_subscription_id_a: NodeAChoice
+        node_subscription_id_b: NodeBChoice
```
Comment (Mark90) [inline workflows/core_link/create_core_link.py]: Note to selves: add to migration guide

File: products/product_blocks/core_link.py [hunk 5]
```diff
@@ -30,9 +28,7 @@
 T = TypeVar("T", covariant=True)
```
Comment (Mark90) [inline products/product_blocks/core_link.py:24]: This TypeVar is unused

File: products/product_blocks/core_link.py [hunk 6]
```diff
@@ -30,9 +28,7 @@
...
 
 
-class ListOfPorts(SubscriptionInstanceList[T]):
-    min_items = 2
-    max_items = 2
+ListOfPorts = Annotated[list[SI], Len(min_length=2, max_length=2)]
```
Comment (Mark90) [inline products/product_blocks/core_link.py:27]: `Len(min_length=2, max_length=2)` could be written as `Len(2, 2)` as well, or `MinLen(2), MaxLen(2)`. @mrijk @hanstrompert  do you have a preference?

File: workflows/core_link/create_core_link.py (hunk 1 above)
Comment (Mark90) [reply workflows/core_link/create_core_link.py]: A model_validator seems like the best solution

File: workflows/l2vpn/shared/forms.py [hunk 7]
```diff
@@ -12,14 +12,14 @@
...
 from orchestrator.types import SubscriptionLifecycle
+from pydantic_forms.validators import Choice, choice_list
 
 from products.product_blocks.port import PortMode
-from workflows.shared import subscriptions_by_product_type_and_instance_value
+from workflows.shared import AllowedNumberOfL2vpnPorts, subscriptions_by_product_type_and_instance_value
 
 
-def ports_selector(number_of_ports: int) -> list:
+def ports_selector(number_of_ports: AllowedNumberOfL2vpnPorts) -> type[list[Choice]]:
```
Comment (Mark90) [inline workflows/l2vpn/shared/forms.py:22]: The annotation `number_of_ports: AllowedNumberOfL2vpnPorts` doesn't hurt, but it also doesn't validate anything. If that's what you expected, you could consider using [`@validate_call`](https://docs.pydantic.dev/latest/concepts/validation_decorator/)

We haven't used this in the SURF orchestrator, it is a nice idea actually.

File: workflows/node/shared/forms.py [hunk 8]
```diff
@@ -34,9 +34,14 @@ def node_type_selector(node_type: str) -> Choice:
...
 
 def node_status_selector() -> Choice:
     statuses = NodeStatus.list()
-    return Choice("NodeStatusEnum", zip(statuses, statuses))  # type:ignore
+    return Choice("NodeStatusEnum", zip(statuses, statuses))
+
+
+SiteChoice: TypeAlias = cast(type[Choice], site_selector())
+NodeRoleChoice: TypeAlias = cast(type[Choice], node_role_selector())
+NodeStatusChoice: TypeAlias = cast(type[Choice], node_status_selector())
```
Comment (Mark90) [inline workflows/node/shared/forms.py:47]: If new netbox entities are created they wont show up in these choices until user restarts the app. Not sure if that's intended?

File: workflows/port/shared/forms.py [hunk 9]
```diff
@@ -10,13 +10,16 @@
...
 
 
-def port_mode_selector() -> list:
+def port_mode_selector() -> Choice:
     port_modes = [port_mode.value for port_mode in PortMode]
-    return Choice("PortModesEnum", zip(port_modes, port_modes))  # type: ignore
+    return Choice("PortModesEnum", zip(port_modes, port_modes))
+
+
+PortModeChoice: TypeAlias = cast(type[Choice], port_mode_selector())
```
Comment (Mark90) [inline workflows/port/shared/forms.py:25]: If new netbox entities are created they wont show up in these choices until user restarts the app. Not sure if that's intended?

File: workflows/shared.py [hunk 10]
```diff
@@ -107,34 +113,28 @@ def free_port_selector(node_subscription_id: UUIDstr, speed: int, enum: str = "P
...
+def summary_form(product_name: str, summary_data: SummaryData) -> Generator:
+    ProductSummary: TypeAlias = cast(type[MigrationSummary], migration_summary(summary_data))
 
     class SummaryForm(FormPage):
-        class Config:
-            title = f"{product_name} summary"
+        model_config = ConfigDict(title=f"{product_name} summary")
 
-        product_summary: ProductSummary
+        product_summary: ProductSummary  # type: ignore
```
Comment (Mark90) [inline workflows/shared.py:136]: If the typealias isn't working, might as well not use a typealias. I'll try running mypy to see what the problem is

Comment (Mark90) [reply workflows/shared.py:136]: The type ignore can be removed

**Comment by Mark90** [comment]
There are still quite a few deprecated `.dict()` calls in the codebase that should be replaced with `.model_dump()`

File: workflows/node/shared/forms.py (hunk 8 above)
Comment (hanstrompert) [reply workflows/node/shared/forms.py:47]: I noticed when I was verifying the workflows 😊 I moved the generation of all dynamic types back inside the input form generator(s).

File: products/product_blocks/core_link.py (hunk 6 above)
Comment (hanstrompert) [reply products/product_blocks/core_link.py:27]: I do not ike `Len(2, 2)`. The other two are fine.

File: workflows/l2vpn/shared/forms.py (hunk 7 above)
Comment (hanstrompert) [reply workflows/l2vpn/shared/forms.py:22]: I added the more specific type to inform the reader of the code, not sure if it makes sense to actually validate it here and risk a exception.

**Pull request merged**